### PR TITLE
Logging cluster labels

### PR DESCRIPTION
**Pull request by mburke5678**
https://issues.redhat.com/browse/OSDOCS-1213
https://issues.redhat.com/browse/LOG-825
https://issues.redhat.com/browse/LOG-717

### CODE REVIEW

**Comment by mburke5678** [comment]
@igor-karpukhin PTAL

**Comment by mburke5678** [comment]
@sichvoge Do you this level of information is sufficient or do we need more?

**Comment by mburke5678** [comment]
@igor-karpukhin PTAL

**Comment by mburke5678** [comment]
@QiaolingTang Can you please review this PR for me? Thank you!

File: modules/cluster-logging-collector-log-forward-es.adoc [hunk 1]
```diff
@@ -45,23 +45,28 @@ spec:
...
 <6> If using an `https` prefix, you must specify the name of the secret required by the endpoint for TLS communication. The secret must exist in the `openshift-logging` project and must have keys of: *tls.crt*, *tls.key*, and *ca-bundle.crt* that point to the respective certificates that they represent.
-<7> Optional. Specify a name for the pipeline.
+<7> Optional: Specify a name for the pipeline.
 <8> Specify which log types should be forwarded using that pipeline: `application,` `infrastructure`, or `audit`.
 <9> Specify the output to use with that pipeline for forwarding the logs.
-<10> Optional. Specify the `default` output to send the logs to the internal Elasticsearch instance.
-<11> Optional. Configure multiple outputs to forward logs to other external Elasticsearch instances.
+<10> Optional: Specify the `default` output to send the logs to the internal Elasticsearch instance.
+<11> Optional: A label to add to the logs.
```
Comment (QiaolingTang) [inline modules/cluster-logging-collector-log-forward-es.adoc:68]: @mburke5678 we can add several labels here, so `A label` is not accurate.
The labels can be added like:
```
labels:
  name1: value1
  name2: value2
  ...
```

File: modules/cluster-logging-collector-log-forward-fluentd.adoc [hunk 2]
```diff
@@ -59,7 +63,8 @@ spec:
 <8> Specify which log types should be forwarded using that pipeline: `application,` `infrastructure`, or `audit`.
 <9> Specify the output to use  with that pipeline for forwarding the logs.
 <10> Optional. Specify the `default` output to forward logs to the internal Elasticsearch instance.
-<11> Optional. Configure multiple outputs to forward other log types to other external log aggregators of any supported type.
+<11> Optional. A label to add to the logs.
```
Comment (QiaolingTang) [inline modules/cluster-logging-collector-log-forward-fluentd.adoc:66]: Same here

File: modules/cluster-logging-collector-log-forward-kafka.adoc [hunk 3]
```diff
@@ -41,29 +41,36 @@ spec:
...
 <4> Specify the `kafka` type.
 <5> Specify the URL and port of the Kafka broker as a valid absolute URL, optionally with a specific topic. You can use the `tcp` (insecure) or `tls` (secure TCP) protocol. If the cluster-wide proxy using the CIDR annotation is enabled, the output must be a server name or FQDN, not an IP address.
 <6> If using a `tls` prefix, you must specify the name of the secret required by the endpoint for TLS communication. The secret must exist in the `openshift-logging` project and must have keys of: *tls.crt*, *tls.key*, and *ca-bundle.crt* that point to the respective certificates that they represent.
-<7> Optional. Specify a name for the pipeline.
+<7> Optional: Specify a name for the pipeline.
 <8> Specify which log types should be forwarded using that pipeline: `application,` `infrastructure`, or `audit`.
 <9> Specify the output to use with that pipeline for forwarding the logs.
-<10> Optional. Configure multiple outputs to forward logs to other external log aggregtors of any supported type.
-<11> Specify `default` to forward logs to the internal Elasticsearch instance.
+<10> Optional: A label to add to the logs.
```
Comment (QiaolingTang) [inline modules/cluster-logging-collector-log-forward-kafka.adoc:71]: Same issue here.

File: modules/cluster-logging-collector-log-forward-syslog.adoc [hunk 4]
```diff
@@ -69,11 +73,12 @@ spec:
...
 <7> If using a `tls` prefix, you must specify the name of the secret required by the endpoint for TLS communication. The secret must exist in the `openshift-logging` project and must have keys of: *tls.crt*, *tls.key*, and *ca-bundle.crt* that point to the respective certificates that they represent.
-<8> Optional. Specify a name for the pipeline.
+<8> Optional: Specify a name for the pipeline.
 <9> Specify which log types should be forwarded using that pipeline: `application,` `infrastructure`, or `audit`.
 <10> Specify the output to use  with that pipeline for forwarding the logs.
-<11> Optional. Specify the `default` output to forward logs to the internal Elasticsearch instance.
-<12> Optional. Configure multiple outputs to forward logs to other external log aggregtors of any supported type.
+<11> Optional: Specify the `default` output to forward logs to the internal Elasticsearch instance.
+<12> Optional: A label to add to the logs.
```
Comment (QiaolingTang) [inline modules/cluster-logging-collector-log-forward-syslog.adoc:80]: Same issue here.

File: modules/cluster-logging-collector-log-forward-es.adoc [hunk 5]
```diff
@@ -45,23 +45,28 @@ spec:
...
-   - name: infrastructure-audit-logs <11>
+     labels:
+       logs: application <11>
+   - name: infrastructure-audit-logs <12>
      inputRefs:
      - infrastructure
      outputRefs:
      - elasticsearch-insecure
+     labels:
+       logs: audit-infra
```
Comment (kalexand-rh) [inline modules/cluster-logging-collector-log-forward-es.adoc:56]: I might add the annotation to this line, too, or add expand the description to acknowledge that each pipeline definition contains an optional label. (Please apply any changes that you make to this YAML to the other YAML examples that you've changed, except the last one.)

File: modules/cluster-logging-collector-log-forwarding-about.adoc [hunk 6]
```diff
@@ -32,6 +32,8 @@ If the output URL scheme requires TLS (HTTPS, TLS, or UDPS), then TLS server-sid
 
 * `audit`. Logs generated by the node audit system (auditd) and the audit logs from the Kubernetes API server and the OpenShift API server. 
 --
++
+You can add labels to outbound log messages using `key:value` pairs in the pipeline. For example, you could add a label to messages forwarded to others data centers or label the logs by type. Labels added to objects are also forwarded with the log message.
```
Comment (kalexand-rh) [inline modules/cluster-logging-collector-log-forwarding-about.adoc:36]:
```suggestion
You can add labels to outbound log messages by using `key:value` pairs in the pipeline. For example, you might add a label to messages that are forwarded to others data centers or label the logs by type. Labels that are added to objects are also forwarded with the log message.
```

**Review by kalexand-rh — COMMENTED** [review]
I have a request and a suggestion. This is looking good. :)

**Pull request merged**